tools/Google.Cloud.Tools.VersionCompat/CecilUtils/SamePropertyComparer.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Tools.VersionCompat.Utils;
+using Mono.Cecil;
+using System.Collections.Generic;
+
+namespace Google.Cloud.Tools.VersionCompat.CecilUtils
+{
+    internal class SamePropertyComparer: IEqualityComparer<PropertyDefinition>
+    {
+        public static IEqualityComparer<PropertyDefinition> Instance { get; } = new SamePropertyComparer();
+        private SamePropertyComparer() { }
+        public bool Equals(PropertyDefinition x, PropertyDefinition y)
+        {
+            if (x is null && y is null)
+            {
+                return true;
+            }
+            if (x is null || y is null)
+            {
+                return false;
+            }
+            if (x.Name != y.Name)
+            {
+                return false;
+            }
+            if (x.Parameters.Count != y.Parameters.Count)
+            {
+                return false;
+            }
+            foreach (var (xParam, yParam) in x.Parameters.Zip(y.Parameters))
```

Comment:
I believe this will consider `out`, `ref` and `in` parameters as equal, which possibly isn't ideal.

Reply:
Ah, I see now it's deliberate. Possibly add a comment explaining this?

Reply:
Comments added to each of the comparers.